### PR TITLE
Refactor schema manager tool

### PR DESCRIPTION
**Pull request by rbans96**
## Description
This PR refactors the schema manager tool code.
The change highlights:

- Moved the code to the SqlServer project which will be shared with the Paas schema management job(We can decide later if we need to create a separate project just for this shared code)
- Updated static classes to non-static classes to support Dependency injection and testability
- Moved Routing classes(KnownRoutes and RouteNames) from sqlServer.Api package to SqlServer to reuse routes in SchemaClient.cs

## Related issues
Addresses [AB#79268
](https://microsofthealth.visualstudio.com/Health/_boards/board/t/Resolute/Stories/?workitem=79268)
## Testing
Manual testing of apply, current and available command with the updated tool.

### CODE REVIEW

File: src/Microsoft.Health.SqlServer/Features/Schema/Manager/ISchemaManagerDataStore.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Health.SqlServer.Features.Schema.Manager
+{
+    public interface ISchemaManagerDataStore
```
Comment (YazanMSFT) [inline src/Microsoft.Health.SqlServer/Features/Schema/Manager/ISchemaManagerDataStore.cs:11]: Could you add comments for the methods here and inheritdoc in SchemaManagerDataStore?

File: src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaManagerDataStore.cs [hunk 2]
```diff
@@ -7,21 +7,27 @@
...
-        public static async Task ExecuteScriptAndCompleteSchemaVersionAsync(string connectionString, string script, int version, CancellationToken cancellationToken)
+        public SchemaManagerDataStore(ISqlConnectionFactory sqlConnectionFactory)
         {
-            using (var connection = new SqlConnection(connectionString))
+            EnsureArg.IsNotNull(sqlConnectionFactory);
+
+            _sqlConnectionFactory = sqlConnectionFactory;
+        }
+
+        public async Task ExecuteScriptAndCompleteSchemaVersionAsync(string script, int version, CancellationToken cancellationToken)
```
Comment (YazanMSFT) [inline src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaManagerDataStore.cs:28]: Could you add parameter validation to public methods?

**Review by YazanMSFT — APPROVED** [review]

File: src/Microsoft.Health.SqlServer/Features/Schema/Manager/ISchemaManagerDataStore.cs (hunk 1 above)
Comment (rbans96) [reply src/Microsoft.Health.SqlServer/Features/Schema/Manager/ISchemaManagerDataStore.cs:11]: Sure, done

File: src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaManagerDataStore.cs (hunk 2 above)
Comment (rbans96) [reply src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaManagerDataStore.cs:29]: Done

**Pull request merged**